app/assets/javascripts/components/articles/article_graphs.jsx

```diff
@@ -1,170 +1,148 @@
-import React from 'react';
-import createReactClass from 'create-react-class';
-import PropTypes from 'prop-types';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import Wp10Graph from './wp10_graph.jsx';
 import EditSizeGraph from './edit_size_graph.jsx';
 import Loading from '../common/loading.jsx';
-import request from '../../utils/request';
+import request from '../../utils/request.js';
 
+const ArticleGraphs = ({ article }) => {
+  const { id: article_id } = article;
 
-const ArticleGraphs = createReactClass({
-  displayName: 'ArticleGraphs',
+  const [showGraph, setShowGraph] = useState(false);
+  const [selectedRadio, setSelectedRadio] = useState('wp10_score');
+  const [articleData, setArticleData] = useState(null);
 
-  propTypes: {
-    article: PropTypes.object
-  },
+  const elementRef = useRef(null);
 
-  getInitialState() {
-    return {
-      showGraph: false,
-      selectedRadio: 'wp10_score',
-      articleData: null
-    };
-  },
-
-  componentDidMount() {
-    this.ref = React.createRef();
-  },
-
-  componentDidUpdate(_, prevState) {
-    if (this.state.showGraph && !prevState.showGraph) {
-      // Add event listener when the component is visible
-      document.addEventListener('mousedown', this.handleClickOutside);
+  useEffect(() => {
+    if (showGraph) {
+      document.addEventListener('mousedown', handleClickOutside);
+    } else {
+      document.removeEventListener('mousedown', handleClickOutside);
```

Comment:
this won't actually work because `handleClickOutside` is defined within this function itself - which means that it will be recreated everytime the component rerenders. The reason it worked earlier(with the class component) was because only the render method is called during rerenders and so `this.handleClickOutside` never changed

Reply:
so i dont understand when you say it wont work. are you saying that because of performance reason we should only create the `handleClickOutside` once through out the lifecycle of the component?

because yes it does work and I can click outside to close the graph

@TheTrio what do you think the solution should be?

Reply:
it will work - the problem is that it won't remove the old event listener. You can verify this by putting a console.log in `handleClickOutside` and then going through this cycle of hiding and showing the graph. You will get an extra console log every time since the old listeners aren't cleaned up